setup.py

```diff
@@ -39,7 +39,7 @@
         'threadloop>=1,<2',
         'thrift',
         'tornado>=4.3,<5',
-        'opentracing>=1.2.2,<2',
+        'opentracing>=2.0,<2.1',
```

Comment:
why not `<3`?

Reply:
If it can be tested on 1.2.2 the lower bound could be kept.  This aids people migrating to the new version as well.

Reply:
Out of cautiousness, but yeah - minors shouldn't break it so I'm fine with `<3`.

Reply:
I could try testing with `1.2.2` and it would probably be doable to pass for both versions, but I think it would be confusing since the jaeger-client code would suggest that scope managers work, whereas in reality that would depend on the opentracing version you're using. I'd rather go with `>=2.0,<3.0`.

Reply:
In that case keep in mind that you will not be co-installable with your previous version.  This could mean you would not be co-installable with projects you previously were (other projects that included opentracing as a dependency).  The only way to update would be to have the OTHER project update as well, everything being in lockstep.

Reply:
There's this code in the Tracer
```
        if self.active_span is not None and not ignore_active_span:
            parent = self.active_span
```

I assume `self.active_span` is coming from the base class, so it won't work without 2.x

Reply:
@prometheanfire it's a fair point but making it compatible with both `2.x` and `1.x` will result in quite a bit of funky code with some `except ImportError` and `getattr` here and there, which makes for a less readable / maintainable code. Also the tests would need to be ran for both versions. @yurishkuro, @black-adder what do you think - is it worth the effort and consequences? I'm fine with forcing people to upgrade their other dependencies when they want to use a newer jaeger-client.

Reply:
For me, I'd prefer to just make a breaking change and release jaeger-client-python 4.0.0 along with some other long outstanding issues: https://github.com/jaegertracing/jaeger-client-python/issues/205

Reply:
+1, but "Remove tornado dependency #31" might be a tall order. Then again, without it  it's difficult to clean up the API so that ioloop is completely hidden.